mv/MBS_MobileDirPad.js

```diff
@@ -221,64 +567,156 @@ MBS.MobileDirPad = {};
 	}
 
-	Sprite_Button.prototype = Object.create(Sprite_Base.prototype);
-	Sprite_Button.prototype.constructor = Sprite_Button;
+	Sprite_mbsButton.prototype = Object.create(Sprite_Base.prototype);
+	Sprite_mbsButton.prototype.constructor = Sprite_mbsButton;
 
-	Sprite_Button.prototype.initialize = function(type) {
+	Sprite_mbsButton.prototype.initialize = function(type) {
 		Sprite_Base.prototype.initialize.call(this);
 		this._type = type;
-		if ((type == 0 ? $.Param.button : $.Param.cButton) == "") this.visible = false;
-		else
-			this.bitmap = ImageManager.loadNormalBitmap(type == 0 ? $.Param.button : $.Param.cButton, 0);
+		
+		let imgPath = '';  // default, empty string
+	switch (type) {
+		case 0:
+	imgPath = MBS.MobileDirPad.Param.actionButtonImage;
+    break;
+		case 1:
+    imgPath = MBS.MobileDirPad.Param.cancelButtonImage;
+    break;
+		case 2:
+    imgPath = MBS.MobileDirPad.Param.shiftButtonImage;
+    break;
+		case 3:
+    imgPath = MBS.MobileDirPad.Param.menuButtonImage;
+    break;
+		case 4:
+    imgPath = MBS.MobileDirPad.Param.qButtonImage;
+    break;
+		case 5:
+    imgPath = MBS.MobileDirPad.Param.wButtonImage;
+    break;
+		default:
+    console.error('MBS_MobileDirPad_Upgraded.js: button type ' + type + ' not recognised!');
+    break;
+}
+		// These next two lines are the ones you edited
+		if (imgPath == '') {
+			this.visible = false;  // make invisible if no image is assigned
+		}else{
+			this.bitmap = ImageManager.loadSystem(imgPath);  // else load image
+		}
+		 
 
-		this.anchor.y = 0.5;
-		this.anchor.x = 0.5;
+		this.anchor.y = 0;//0.5;
+		this.anchor.x = 0;//0.5;
 		this._moveDuration = 0;
 		this._moveSpeed = 0;
 		this.z = 5;
 	};
 
-	Sprite_Button.prototype.update = function() {
+	Sprite_mbsButton.prototype.update = function() {
 		Sprite_Base.prototype.update.call(this);
 		if (!this.visible) return;
 		this.updateMovement();
 		this.updateTouch();
 	};
 
-	Sprite_Button.prototype.updateMovement = function() {
+	Sprite_mbsButton.prototype.updateMovement = function() {
 		if (this._moveDuration > 0) {
 			this.x += this._moveSpeed;
 			this._moveDuration--;
 		}
 	};
 
-	Sprite_Button.prototype.updateTouch = function() {
-		if (this._type == 0 && TouchInput.isPressed()) {
-			var rect = new PIXI.Rectangle(this.x - this.width * this.anchor.x, this.y - this.height * this.anchor.y, this.width, this.height);
-			Input._currentState['ok'] = rect.contains(TouchInput.x, TouchInput.y);
-		} else if (this._type == 0) {
-			Input._currentState['ok'] = false;
-		} else if (this._type == 1 && TouchInput.isTriggered()) {
-			var rect = new PIXI.Rectangle(this.x - this.width * this.anchor.x, this.y - this.height * this.anchor.y, this.width, this.height);
-			Input._currentState['escape'] = rect.contains(TouchInput.x, TouchInput.y);
-		} else if (this._type == 1) {
-			Input._currentState['escape'] = false;
+	Sprite_mbsButton.prototype.updateTouch = function() {
+		switch(this._type){
```

Comment:
OK, this one is real bad. I know the original was like that (kind of) but this whole switch statement is a clear code smell.

I think you can apply the same logic i suggested on the other switch clause and have this reduced to something like, 5 lines of code.